pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -409,20 +414,15 @@ PJ_DEF(pj_status_t) pjmedia_sdp_neg_modify_local_offer2(
     }
 
     /* New_offer fixed */
-#if PJMEDIA_SDP_NEG_COMPARE_BEFORE_INC_VERSION
     new_offer->origin.version = old_offer->origin.version;
 
-    if (pjmedia_sdp_session_cmp(new_offer, neg->initial_sdp, 0) != PJ_SUCCESS)
-    {
+    if (pjmedia_sdp_session_cmp(neg->prev_sent, new_offer, 0) != PJ_SUCCESS) {
         ++new_offer->origin.version;
-    }    
-#else
-    new_offer->origin.version = old_offer->origin.version + 1;
-#endif
-    
+    }
     neg->initial_sdp_tmp = neg->initial_sdp;
     neg->initial_sdp = new_offer;
     neg->neg_local_sdp = pjmedia_sdp_session_clone(pool, new_offer);
+    neg->prev_sent = neg->neg_local_sdp;
```

Comment:
Just a note, the SDP nego module will use a flip-flop pools setup. The pool used for cloning the SDP is usually a provisional pool, which means it may be released/reset after SDP nego if the nego fails. When the nego succeeds, it will be an active pool (will not be reset until a next successful SDP nego). That's why maintaining the `initial_sdp` looks a bit weird. Didn't checked the code detail, but please make sure it is safe in such flip-flop pool setup :)

Reply:
So is this handled already? Because it seems that there's no reversion for `last_sent` here:
https://github.com/pjsip/pjproject/blob/caabba58a0c27229998a510b7aa85832623493ae/pjmedia/src/pjmedia/sdp_neg.c#L1631
Shouldn't it be the same like `initial_sdp`?

And btw, what is `initial_sdp` now used for? i.e. what's the difference between `initial_sdp` and `last_sent` now?

Reply:
Actually reversion may not be needed for `last_sent` as whatever has been sent to remote cannot be reversed/cancelled? Also the provisional pool (`pool_prov`) will not be reset after a failed nego, so it looks safe already in the flip-flop pool setup.

Re: `initial_sdp`, it seems to be used for storing initial local capability which is used by `pjmedia_sdp_neg_set_local_answer()`
https://github.com/pjsip/pjproject/blob/c43af43627baeb2caa43977f6783081906bd1908/pjmedia/include/pjmedia/sdp_neg.h#L651-L657

Reply:
But in `sip_inv.c` `inv_negotiate_sdp()`, there's something like this:
```
    status = pjmedia_sdp_neg_negotiate(inv->pool_prov, inv->neg, 0);
...
    if (inv->state != PJSIP_INV_STATE_DISCONNECTED) {

        /* Swap the flip-flop pool when SDP negotiation success. */
        if (status == PJ_SUCCESS) {
            swap_pool(&inv->pool_prov, &inv->pool_active);
        }

        /* Reset the provisional pool regardless SDP negotiation result. */
        pj_pool_reset(inv->pool_prov);
```
So if `last_sent` is allocated using pool_prov, it may have been reset?

Reply:
Ah you're right, missed that! Thought the outer `if` is SDP nego result check.